### ISSUE TITLE
Build starts with base branch even if the job is PR specific job.

### ISSUE THREAD

**Issue by s-yoshika** [screwdriver-cd/screwdriver#1760]
**What happened**:
Given the pipeline whose base branch is **master** and whose jobs is like below.
```
jobs:
  test:
    requires: [ ~pr:foo ]
    image: node:10
    steps:
    - echo: echo ok      
```
The `test` job is triggered by specific branch **foo**, and the PR build check out from **foo** branch and merge it with PR branch. This is correct behavior.
But, next if it restarted by manual in SD UI, it check out from **master** branch.

**What you expected to happen**:
The PR build of branch specific job always check out from the specific branch(**foo**) , even when it restarted in SD UI.

**How to reproduce it**:
1. Make pipeline by master branch.
2. Define PR branch filtering job .
3. Create foo branch.
4. Create bar branch
5. Create new PR from bar to foo.
6. The job is executed with foo branch.
7. Restart the PR job in SD UI
8. The job is executed with master branch.

**Comment by tk3fftk** [comment]
`startFrom` of a restarted event is different from a triggered event.
`startFrom` of a restarted event is `PR-1234:job_name` but triggered event's is `~pr:branch_name`.
This causes checkout branch difference in there: https://github.com/screwdriver-cd/scm-base/blob/e34392997dec0f5f1ead31b6468de9ae113274ed/index.js#L175

The first case (restarted) does not matches the regexp so passing pipeline branch to [getCheckoutCommand()](https://github.com/screwdriver-cd/scm-github/blob/fa0d736e95ea4d4fc5747c25f3a6ff91daa8ec5d/index.js#L315).
The second case (triggered) matches the regexp so passing branch name of `~pr:branch_name`.

**Comment by tk3fftk** [comment]
It looks similar issue.
https://github.com/screwdriver-cd/screwdriver/issues/1755

**Comment by jithin1987** [comment]
please close this, in favor of #1755 if this is the same issue

**Comment by marian-r** [comment]
@jithin1987 that is not the same issue. The issue in #1755 happens for subsequent jobs in the pipeline, this issue happens on manual restarts in SD UI. Both seem quite critical to me as the build starts on a different branch than the engineer expects.
